src/printer.js

```diff
@@ -58,13 +58,29 @@ function hasPrettierIgnoreComment(node) {
   );
 }
 
+function isOutsideRange(node, rangeStart, rangeEnd) {
+  if (!node) {
+    return false;
+  }
+  if (node.type === 'File' || node.type === 'Program') {
+    return false;
+  }
+  const nodeStart = util.locStart(node);
+  const nodeEnd = util.locEnd(node);
+  const isOutside = nodeEnd < rangeStart || nodeStart > rangeEnd;
+  return isOutside;
+}
+
 function genericPrint(path, options, printPath, args) {
   assert.ok(path instanceof FastPath);
 
   var node = path.getValue();
 
   // Escape hatch
-  if (hasPrettierIgnoreComment(node)) {
+  if (
+    hasPrettierIgnoreComment(node) ||
+    isOutsideRange(node, options.rangeStart, options.rangeEnd)
```

Comment:
This logic currently isn't clever enough to isolate formatting to nodes within another node. In other words, formatting is applied to parent nodes until a top-level node is reached. This isn't ideal, but it's a start.

Demo: Formatting this code with `prettier --range-start 61 --range-end 81` touches more than just one line:

```js
if ( true ) {
    tooMuchIndentation();
}
if ( true ) {
    tooMuchIndentation();
    // THE ABOVE LINE IS THE ONLY ONE IN THE RANGE 61-81
    // BUT THE LINE ABOVE THAT (`if ( true ) {`) ALSO GETS FORMATTED
}
if ( true ) {
    tooMuchIndentation();
}
```